src/lib/shell/streamer_esp32.cpp

```diff
@@ -44,6 +44,74 @@ static int chip_command_handler(int argc, char ** argv)
     return static_cast<int>(err.AsInteger());
 }
 
+void get_command_completion(const char * buf, linenoiseCompletions * lc)
+{
+    size_t len = strlen(buf);
+
+    // If there is a trailing space, remove it for completion lookup
+    bool trailing_space = false;
+    if (buf[len - 1] == ' ')
+    {
+        trailing_space = true;
+        memset((char *) &buf[len - 1], 0, 1);
+        len -= 1;
+    }
+
+    const char * matter_prefix = "matter";
+
+    // the commands are prefixed with "matter"
+    if (len >= 6 && strncmp(buf, matter_prefix, 6) == 0)
+    {
+        // remove "matter " prefix for completion lookup as the command
+        // registeration in general are done without "matter" prefix
+        char * line = new char[len - strlen(matter_prefix) + trailing_space + 1];
```

Comment:
Please don't use bools in integer arithmentic. Feels like "tricky code" even if it works.

Reply:
Use scoped buffers for line, check allocation.